numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2571,6 +2571,31 @@ void_dealloc(PyVoidScalarObject *v)
     Py_TYPE(v)->tp_free(v);
 }
 
+
+static PyObject *
+object_arrtype_alloc(PyTypeObject *type, Py_ssize_t items)
+{
+    /*
+     * Object scalars should not actually exist, if they exist we should
+     * consider it to be a bug.
+     */
+    static PyObject *visibleDeprecationWarning = NULL;
+    npy_cache_import("numpy", "VisibleDeprecationWarning",
+                     &visibleDeprecationWarning);
+    if (visibleDeprecationWarning == NULL) {
+        return NULL;
+    }
+    if (PyErr_WarnEx(visibleDeprecationWarning,
+            "Creating a NumPy object scalar.  NumPy object scalars should "
+            "never be created.  If you see this message please inform the "
+            "NumPy developers.  Since this message should never be shown "
```

Comment:
Should we link to the issue tracker here ?

Reply:
I think it is fine like this, people who might be doing this should know where to find us.